nixos/modules/misc/nixpkgs.nix

```diff
@@ -280,6 +280,11 @@ in
     system = mkOption {
       type = types.str;
       example = "i686-linux";
+      default =
+        throwIf
+          (opt.system.highestPrio == (lib.mkOptionDefault {}).priority)
```

Comment:
This is rather ugly. It's cleaner to change the type of `system` to `nullOr str` and having a default of `null` and using `assertions`

Reply:
Hmm. The really ugly bit is that `default = throw "..."` did not work as one might expect.
Otherwise I thought that this localized solution was quite on point.

Reply:
It was an issue with the derivation scrubbing logic in the docs. I've simplified this.